R/write.R

```diff
@@ -5,139 +5,205 @@
 #' Default `TRUE`.
-#' @param incl_meta Include a sheet capturing the combined output of the
+#' @param include_metadata Include a sheet capturing the combined output of the
 #' supertibble `redcap_metadata`. Default `TRUE`.
-#' @param tableStyle Any excel table style name or "none" (see "formatting"
-#' vignette in \link[openxlsx]{writeDataTable}). Default "TableStyleLight10".
+#' @param table_style Any excel table style name or "none" (see "formatting"
+#' vignette in \link[openxlsx2]{wb_add_data_table}). Default "tableStyleLight8".
+#' @param set_col_widths Width to set columns across the workbook. Default
+#' "auto", otherwise a numeric value. Standard Excel is 8.43.
+#' @param recode_yn Convert REDCap yesno questions from TRUE/FALSE to "yes"/"no"
+#' for readability. Default TRUE.
+#' @param na_strings Value used for replacing NA values from x. Default
+#' `na_strings()` uses the special ⁠#N/A⁠ value within the workbook.
 #'
 #' @importFrom purrr map map2
-#' @importFrom stringr str_trunc
+#' @importFrom stringr str_trunc str_replace_all str_squish
+#' @importFrom dplyr select pull
+#' @importFrom rlang check_installed
 #'
 #' @return
-#' A workbook object
+#' An `openxlsx2` workbook object
 #'
 #' @examples
 #' \dontrun{
 #' redcap_uri <- Sys.getenv("REDCAP_URI")
 #' token <- Sys.getenv("REDCAP_TOKEN")
 #'
 #' supertbl <- read_redcap(redcap_uri, token)
-#' write_supertibble_xlsx(supertbl, file = "supertibble.xlsx")
+#' write_redcap_xlsx(supertbl, file = "supertibble.xlsx")
 #'
-#' write_supertibble_xlsx(supertbl, labelled = TRUE, file = "supertibble.xlsx")
+#' write_redcap_xlsx(supertbl, labelled = TRUE, file = "supertibble.xlsx")
 #' }
 #'
 #' @export
 
-write_supertibble_xlsx <- function(supertbl,
-                                   labelled = FALSE,
-                                   labelled_sheets = FALSE,
-                                   file,
-                                   incl_supertbl = TRUE,
-                                   incl_meta = TRUE,
-                                   tableStyle = "TableStyleLight10" #nolint: object_name_linter
+write_redcap_xlsx <- function(supertbl,
+                              file,
+                              labelled = NULL,
+                              use_labels_for_sheet_names = TRUE,
+                              include_toc_from_supertbl = TRUE,
+                              include_metadata = TRUE,
+                              table_style = "tableStyleLight8",
+                              set_col_widths = "auto",
+                              recode_yn = TRUE,
+                              na_strings = openxlsx2::na_strings()
 ) {
+  # Enforce checks ----
+  check_arg_is_supertbl(supertbl)
+  check_arg_is_character(table_style, any.missing = FALSE)
+  check_arg_is_character(file, any.missing = FALSE)
+  check_arg_is_valid_extension(file, valid_extensions = c("xlsx"))
+  check_arg_is_character(set_col_widths, any.missing = FALSE)
```

Comment:
I don't think this is right. What if you pass a number?

Reply:
You're right, number is also acceptable.

> width to set cols to specified in Excel column width units or "auto" for automatic sizing. The widths argument is recycled to the length of cols. The default width is 8.43. Though there is no specific default width for Excel, it depends on Excel version, operating system and DPI settings used. Setting it to specific value also is no guarantee that the output will be of the selected width.

Reply:
Do we want to just remove this check or create a new custom one that checks for auto or numeric input? @ezraporter 

I see it was updated to reflect the new `column_width` arg rename

Reply:
Oh I meant to remove, sorry. IMO it's fine to not have any check here since we're just forwarding to `openxlsx2`. Really we want to accept any input that's valid for that package.